### ISSUE TITLE
GSI in online manner and feature extraction from different models?

### ISSUE THREAD

**Issue by abdullahjamal** [dyhBUPT/StrongSORT#42]
Hi, 
Thanks for the great work on the top of deepsort. I have couple of questions regarding GSI and feature extraction which are:

1) For GSI, do I need to store the tracking results from any trackor in .txt file and then apply GSI? or can I use it after I get the tracking results in an online manner?

2) For feature extraction, I see you use ReID models, but if my videos are not RGB and based of grayscale frames. Is is possible to use the detector model as feature extractor? What do you suggest which model might be good enough for appearance feature. I don't have ground truth person ids so I want the trackor to work in an online setting.

**Comment by dyhBUPT** [comment]
Hi, thanks for your interest in our work.

1. For GSI, for every, you must give it a complete trajectory as input. You it's hard to work in an online manner. BTW, both AFLink and GSI are all offline modules.
2. For feature extraction, the feature from detector is not very suitable for ReID. Maybe you can choose a ReID dataset, then change the images of it to grayscales, and train the ReID model.

Best wishes.

**Comment by abdullahjamal** [comment]
Thanks for the response.

1) What do you suggest how can I use GSI? Let's assume I'm using Bytetrack or BoTSORT as a trackor. Secondly, do I need to train AFLink first separately?
2) That's an interesting approach.  Sometimes the camera might be different, and in turns the input. It might not ideally be grayscale.

**Comment by dyhBUPT** [comment]
1. Maybe you can apply GSI to the trajectory whenever it dies. On your own dataset, maybe you should retrain the AFLink and finetune the hyperparameters.
2. Oh, that seems bad. Maybe you can refer to [FairMOT](https://github.com/ifzhang/FairMOT), which tries to train ReID model without id labels.

**Comment by abdullahjamal** [comment]
Thanks for the suggestions.

1- How do you create a groundtruths for AFLink?
2- I will try to look into FairMOT

**Comment by dyhBUPT** [comment]
Hi, for gt of AFLink, you can delve into the codes https://github.com/dyhBUPT/StrongSORT/blob/b6e981c94448c9d3be7f718ef23ebada1324ec3e/AFLink/dataset.py#L38

You're also welcomed to find the problems of AFLink and propose improvements.

Best wishes.
